mobile/native/src/trade/order/orderbook_client.rs

```diff
@@ -15,10 +18,19 @@ impl OrderbookClient {
     }
 
     pub(crate) async fn post_new_order(&self, order: NewOrder) -> Result<OrderResponse> {
+        let secret_key = get_node_key();
+        let message = order.message();
+        let signature = secret_key.sign_ecdsa(message);
```

Comment:
🔧 It would probably reasonable to put the signing into a function and do not leak the secret_key just like that. Have a look at the `AesCipher`.

Reply:
Where is the advantage? The `AesCipher` needs to be initialized using the `secret_key`, meaning, we just move this to a different place.

Reply:
@luckysori : I'm keen on hearing your thoughts on this and happily will change it in a follow-up PR.